valve-monitor/web-app/src/services/AppModel.ts

```diff
@@ -38,3 +38,16 @@ export interface ProjectDataSnapshot {
   when: AppDate;
   project: Project;
 }
+
+export interface ValveMonitorDevice {
+  deviceID: string;
+  name: string;
+  lastActivity: string;
+  valveState: string;
+  deviceFlowRate: number | string;
```

Comment:
I’d pick one or the other. It’s kind of weird to allow both in my opinion.

Reply:
It's there just so the default if a number's not provided can be "x"